### DISCUSSION TITLE
NEED ASSISTANCE? Click here for our AMR for R Assistant chatbot!

### DISCUSSION THREAD

**Discussion by msberends** [msberends/AMR#18] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

## AMR for R Assistant 

<a target="_blank" href="https://chatgpt.com/g/g-M4UNLwFi5-amr-for-r-assistant"><img src="https://github.com/msberends/AMR/raw/main/pkgdown/assets/AMRforRGPT.svg"></a>

https://chat.amr-for-r.org

**Comment by ghost** [comment]
Hello, i would first like to thank all the team working on AMR package, this is wonderful.

As I am quite new to coding in R, I would like to have some help. I am currently working with an antimicrobial resistance dataset, focusing on multi drug resistance patterns in methicillin resistant staphylococcus aureus (MRSA).

1) is it possible to differentiate and get the exact count between MDR, XDR and PDR, using the CMI2012 guidelines?

2) concerning the MRSA, termed as such when resistant (R) to cefoxitin (FOX), is it possible to pool the proportion of resistance of MRSA to other antibiotics (in other word this is to specifically answer the question: what proportion of MRSA are drug/multi drug resistant ?)

I am working with a standard dataset with column headers as: patient_id, service, bacteria, and the rest of the columns containing susceptibility/resistance data of antibiotics.

**Reply by msberends** [reply]
Hi! Thank you so much for using the package and letting know that you find it helpful - that’s the sole reason we developed it 😊

Sure, your use case is probably quite common and indeed supported by the AMR package. We wrote an article on the determination of MDR/XDR/PDR: https://msberends.github.io/AMR/articles/MDR.html. In short, you can use the [`mdro()` function](https://msberends.github.io/AMR/reference/mdro.html) to determine multi drug resistance. 

You can just add an MDRO column like this:

```r
# using tidyverse/dplyr 
library(dplyr)
library(AMR)
your_data %>%
  mutate(MDRO = mdro(col_mo = "bacteria"))

# using base R
your_data$MDRO <- mdro(your_data,
                       col_mo = "bacteria")
```

Now you can retrieve the counts you requested for using either of these methods:

```r
# using tidyverse/dplyr 
library(dplyr)
library(AMR)
your_data %>%
  count(MDRO)

# using base R
table(your_data$MDRO)
```

To answer both your questions, you have to filter first before you use the `count()` or `table()` function. For you first question, you should filter on S. aureus isolates, for your second question you can afterwards filter on oxa or fluclox resistant isolates. 

If you find it hard to filter, please let me know! I can add some examples for you.

**Reply by ghost** [reply]
thank you for your response. And yes, I will find it very helpul if you could provide me with an example on how to filter. 

thank you again

**Reply by theanita1** [reply]
Hi ghost, 
For your question on MRSA, as with anything in R there's multiple ways to do it - here's a few that I've used 
1) create a specific MRSA subset and then use that to calculate the proportions for the individual antibiotics: 
```
mrsa_df<- subset(df, mo_column == "B_STPHY_AURS", FOX == "R)
```
#change name of 'mo_col to whatever you have as your column for your organisms, and 'FOX' for the column name where you have cefoxitin and/or oxacillin. You could exchange these two aspects for a simple filter on your MRSA column if you have one

 2) using dplyr 
for the below code, change "group_var" for the column name you have for a column that has a binary variable for MRSA/MSSA, and abx for the column name of the antibiotic you want to know the resistance of

```
df |>
    filter(mo_column == "B_STPHY_AURS") |>
    group_by(group_var) |>
    summarise(tested = n_sir(abx),
              resistant = count_R(abx),
              prop_r = proportion_R(abx, as_percent = T, minimum = 1),
              ci_min = sir_confidence_interval(abx, ab_result = "R", side = "min", as_percent = T, minimum = 1),
              ci_max = sir_confidence_interval(abx, ab_result = "R", side = "max", as_percent = T, minimum = 1))
```

The output is something like this:
![image](https://github.com/user-attachments/assets/3d942081-c3a2-48fb-af37-d286a88bde5b)

If you're doing this calculation a lot, it might be worth making it into a function to shorten your code

**Comment by sherubpvet** [comment]
Hi, How do we calculate the MDR using "The CLSI guideline? "

**Reply by msberends** [reply]
Hi! I’m not sure I follow - is there a CLSI specific guideline or definition for MDRs? I might be a bit oblivious here coming from Europe where EUCAST is mostly used. We don’t have CLSI specific definitions in the package at the moment, but could add these if they exist. 

You can now determine MDROs in these ways: https://msberends.github.io/AMR/reference/mdro.html

I’m happy to help if any help is needed.

**Reply by theanita1** [reply]
HI Sherubpvet
Is the CLSI defition any difference to the Magiorakos definition of
- MDR = resistance to three or more categories
- XDR = susceptible to two or less categories
- PDR  = resistant to all categories

https://pubmed.ncbi.nlm.nih.gov/21793988/

**Comment by theanita1** [comment]
Hey Matthijs,
Wanted to give feedback on the AMR chatGPT - has been useful as I've been updating my previous cleaning code with the new dplyr functions e.g. coalese, mutate(across())
Cheers
